aws/resource_aws_ssm_parameter.go

```diff
@@ -124,24 +169,47 @@ func resourceAwsSsmParameterPut(d *schema.ResourceData, meta interface{}) error
 	log.Printf("[INFO] Creating SSM Parameter: %s", d.Get("name").(string))
 
 	paramInput := &ssm.PutParameterInput{
-		Name:      aws.String(d.Get("name").(string)),
-		Type:      aws.String(d.Get("type").(string)),
-		Value:     aws.String(d.Get("value").(string)),
-		Overwrite: aws.Bool(d.Get("overwrite").(bool)),
+		Name:           aws.String(d.Get("name").(string)),
+		Type:           aws.String(d.Get("type").(string)),
+		Value:          aws.String(d.Get("value").(string)),
+		Overwrite:      aws.Bool(shouldUpdateSsmParameter(d)),
+		AllowedPattern: aws.String(d.Get("allowed_pattern").(string)),
 	}
+
+	if description, ok := d.GetOk("description"); ok {
+		paramInput.SetDescription(description.(string))
+	} else if d.HasChange("description") {
+		// There is a "bug" in the AWS API and is it not possible to unset a description once
```

Comment:
Is this still the case? It sounds like it will cause a perpetual diff if this logic is left as-is (e.g. always showing `description "" => " "` when its removed until a configuration is re-updated to `description = " "` instead). What happens if you set `description = ""` in your configuration?

In either case if the AWS API bug still exists, we should instead force a new resource and document the behavior in the resource documentation.